### PR TITLE
I will add the model class for persistence in the Book class

### PR DESCRIPTION
**Pull request by pablohincapie**
## Summary
Model Class for persistence

## Trello Card
https://trello.com/c/wvAJwo3n/14-crear-modelo-de-book

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @Column(nullable = false)
+    private String publisher;
+    @Column(nullable = false)
+    private String year;
+    @Column(nullable = false)
+    private Integer pages;
+    @Column(nullable = false)
+    private String isbn;
+
+    Book() {
```
Comment (AleFalcon) [inline src/main/java/wolox/training/models/Book.java]: In this line is missing the word "public" for identify that it is a constructor void.
Also missing the constructor with params

**Review by AleFalcon — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (pablohincapie) [reply src/main/java/wolox/training/models/Book.java]: I added a new constructor with params and the public access inn to the exist constructor

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package wolox.training.models;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public class Book {
+    @Id
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java:14]: Create a field called id of type Long and make it self-incrementing

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    private String genre;
+    @Column(nullable = false)
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java:21]: The @column annotations are required to be able to generate the model in the database.
In this case I would add the @column (nulleable = false) annotation in conjunction with the @NotNull annotation.

The parameter **(nulleable = false)** defines and validates that the field of the table in question is not null. This validation is carried out in persistence, that is, in communication with the database.

@NotNull checks that the field is not null in the java layer, that is, before communicating to the database. This avoids sending invalid data to the database.

```suggestion
    @NotNull 
    @Column(nullable = false)
```
Add this annotation in all the fields that have nulleable = false.

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    private String title;
+    @Column(nullable = false)
+    private String subtitle;
+    @Column(nullable = false)
+    private String publisher;
+    @Column(nullable = false)
+    private String year;
+    @Column(nullable = false)
+    private Integer pages;
+    @Column(nullable = false)
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: In these cases where the fields need to be unique ,I would not just add @Column (nullable = false)
I would also add the following:
@Column (nullable = false, unique = true)

```suggestion
    @NotNull
    @Column (nulleable = false, unique = true)
    private String isbn;
```

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Great job Pablo, I leave you just a few comments 😄

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (pablohincapie) [reply src/main/java/wolox/training/models/Book.java:14]: I added a new Id field

File: src/main/java/wolox/training/models/Book.java (hunk 3 above)
Comment (pablohincapie) [reply src/main/java/wolox/training/models/Book.java:21]: I added @NotNull in the mandatory fields

File: src/main/java/wolox/training/models/Book.java (hunk 4 above)
Comment (pablohincapie) [reply src/main/java/wolox/training/models/Book.java]: I added unique in the isbn field. In this case only isbn.

File: src/main/java/wolox/training/models/Book.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: In Postgres when GenerationType.AUTO is used, it generates an incremental IDs shared between the schema tables. This happens for this database engine. Each database engine responds differently to spring auto-increment. To generate independent IDs for each table use GenerationType.SEQUENCE.

```suggestion
@GeneratedValue (strategy = GenerationType.SEQUENCE, generator = "BOOK_SEQ")            
@SequenceGenerator(name = "BOOK_SEQ", sequenceName = "BOOK_SEQ")
```

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
One last detail 😄

**Review by AleFalcon — COMMENTED** [review]
Great job

File: src/main/java/wolox/training/models/Book.java (hunk 5 above)
Comment (pablohincapie) [reply src/main/java/wolox/training/models/Book.java]: I changed the way of auto-incremental

**Review by AleFalcon — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**